### PR TITLE
statistics: unify log format in package statistics

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Unify log format in package statistics

### What is changed and how it works?
Unify log format in package statistics.

### Check List 

Tests 

 - No code

Code changes

 - None

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: statistics/update.go [hunk 1]
```diff
@@ -391,7 +392,7 @@ func (h *Handle) DumpStatsFeedbackToKV() error {
 func (h *Handle) dumpFeedbackToKV(fb *QueryFeedback) error {
 	vals, err := encodeFeedback(fb)
 	if err != nil {
-		log.Debugf("error occurred when encoding feedback, err: %s", errors.ErrorStack(err))
+		log.Debug("Error occurred when encoding feedback, err: %s", zap.String("error stack", errors.ErrorStack(err)))
```
Comment (winoros) [inline statistics/update.go]:
```suggestion
		log.Debug("Error occurred when encoding feedback", zap.String("error stack", errors.ErrorStack(err)))
```

File: statistics/update.go [hunk 2]
```diff
@@ -739,7 +740,7 @@ func (h *Handle) autoAnalyzeTable(tblInfo *model.TableInfo, statsTbl *Table, sta
 		return false
 	}
 	if needAnalyze, reason := NeedAnalyzeTable(statsTbl, 20*h.Lease, ratio, start, end, time.Now()); needAnalyze {
-		log.Infof("[stats] %s, auto %s now", sql, reason)
+		log.Info("[stats]", zap.String("sql", sql), zap.String("reason", reason))
```
Comment (winoros) [inline statistics/update.go]:
```suggestion
		log.Info("[stats] auto analyze triggered", zap.String("sql", sql), zap.String("reason", reason))
```

File: statistics/update.go [hunk 3]
```diff
@@ -749,7 +750,7 @@ func (h *Handle) autoAnalyzeTable(tblInfo *model.TableInfo, statsTbl *Table, sta
 		}
 		if _, ok := statsTbl.Indices[idx.ID]; !ok {
 			sql = fmt.Sprintf("%s index `%s`", sql, idx.Name.O)
-			log.Infof("[stats] index unanalyzed, auto %s now", sql)
+			log.Info("[stats] index unanalyzed", zap.String("sql", sql))
```
Comment (winoros) [inline statistics/update.go]:
```suggestion
			log.Info("[stats] auto analyze for unanalyzed index triggered", zap.String("sql", sql))
```

File: statistics/update.go [hunk 4]
```diff
@@ -763,7 +764,8 @@ func (h *Handle) execAutoAnalyze(sql string) {
 	dur := time.Since(startTime)
 	metrics.AutoAnalyzeHistogram.Observe(dur.Seconds())
 	if err != nil {
-		log.Errorf("[stats] auto %v failed: %v, cost_time:%vs", sql, errors.ErrorStack(err), dur.Seconds())
+		log.Error("[stats] auto sql failed", zap.String("sql", sql), zap.Duration("cost_time", dur),
```
Comment (winoros) [inline statistics/update.go]:
```suggestion
		log.Error("[stats] auto analyze failed", zap.String("sql", sql), zap.Duration("cost_time", dur),
```

File: statistics/update.go (hunk 1 above)
Comment (qw4990) [reply statistics/update.go]: All suggestions have been accepted.

File: statistics/bootstrap.go [hunk 5]
```diff
@@ -26,15 +27,15 @@ import (
...
+	"go.uber.org/zap"
 )
 
 func (h *Handle) initStatsMeta4Chunk(is infoschema.InfoSchema, tables statsCache, iter *chunk.Iterator4Chunk) {
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
 		physicalID := row.GetInt64(1)
 		table, ok := h.getTableByPhysicalID(is, physicalID)
 		if !ok {
-			log.Debugf("Unknown physical ID %d in stats meta table, maybe it has been dropped", physicalID)
+			log.Debug("Unknown physical ID in stats meta table, maybe it has been dropped", zap.Int64("ID", physicalID))
```
Comment (lamxTyler) [inline statistics/bootstrap.go]: Lower case for the first word?

File: statistics/bootstrap.go [hunk 6]
```diff
@@ -188,14 +189,14 @@ func initStatsBuckets4Chunk(ctx sessionctx.Context, tables statsCache, iter *chu
 			var err error
 			lower, err = d.ConvertTo(ctx.GetSessionVars().StmtCtx, &column.Info.FieldType)
 			if err != nil {
-				log.Debugf("decode bucket lower bound failed: %s", errors.ErrorStack(err))
+				log.Debug("Decode bucket lower bound failed", zap.String("error stack", errors.ErrorStack(err)))
```
Comment (lamxTyler) [inline statistics/bootstrap.go]: `stack` or `errorStack`?

File: statistics/bootstrap.go (hunk 5 above)
Comment (lamxTyler) [reply statistics/bootstrap.go]: And should we use `logutil.Logger`?

File: statistics/bootstrap.go [hunk 7]
```diff
@@ -188,14 +189,14 @@ func initStatsBuckets4Chunk(ctx sessionctx.Context, tables statsCache, iter *chu
...
 				delete(table.Columns, histID)
 				continue
 			}
 			d = types.NewBytesDatum(row.GetBytes(6))
 			upper, err = d.ConvertTo(ctx.GetSessionVars().StmtCtx, &column.Info.FieldType)
 			if err != nil {
-				log.Debugf("decode bucket upper bound failed: %s", errors.ErrorStack(err))
+				log.Debug("Decode bucket upper bound failed", zap.String("error stack", errors.ErrorStack(err)))
```
Comment (eurekaka) [inline statistics/bootstrap.go]: `zap.Error(err)` should have contained the error stack now, so we can directly use `zap.Error(err)` and remove those `errors.ErrorStack(err)` now? @lysu please confirm this.

Comment (lysu) [reply statistics/bootstrap.go]: yes, zap.Error(err) will contain error stack if err is Traced by pingcap/errors

Comment (qw4990) [reply statistics/bootstrap.go]: Updated.
PTAL~

File: statistics/bootstrap.go (hunk 6 above)
Comment (qw4990) [reply statistics/bootstrap.go]: updated.

File: statistics/bootstrap.go (hunk 5 above)
Comment (qw4990) [reply statistics/bootstrap.go]: fixed.

File: statistics/feedback.go [hunk 8]
```diff
@@ -298,7 +301,7 @@ func buildBucketFeedback(h *Histogram, feedback *QueryFeedback) (map[int]*Bucket
 	for _, fb := range feedback.feedback {
 		skip, err := fb.adjustFeedbackBoundaries(sc, &min, &max)
 		if err != nil {
-			log.Debugf("adjust feedback boundaries failed, err: %v", errors.ErrorStack(err))
+			logutil.Logger(context.Background()).Debug("adjust feedback boundaries failed", zap.Error(err))
```
Comment (zz-jason) [inline statistics/feedback.go]: @lamxTyler Should we change its level to ERROR?

File: statistics/update_test.go [hunk 9]
```diff
@@ -58,6 +61,15 @@ func (s *testStatsSuite) TearDownSuite(c *C) {
 	testleak.AfterTest(c)()
 }
 
+func (s *testStatsSuite) registerHook() {
```
Comment (XuHuaiyu) [inline statistics/update_test.go]:
``` go
func (s *testStatsSuite) registerHook() {
    conf := &log.Config{Level: “info”, File: log.FileLogConfig{}}
    _, r,_ := log.InitLogger(conf)
    s.hook = &logHook{r.Core, “”}
    lg := zap.New(s.hook)
    log.ReplaceGlobals(lg, r)
}

func (h *logHook) Write(entry zapcore.Entry, fields []zapcore.Field) error {
    message := entry.Message
    if idx := strings.Index(message, “[stats”); idx != -1 {
        h.results = h.results + message
        for _, f := range fields {
            h.results = h.results + “, ” + f.Key + “: ” + h.field2String(f)
        }
    }
    return nil
}
```

File: statistics/update_test.go [hunk 10]
```diff
@@ -1079,28 +1110,28 @@ func (s *testStatsSuite) TestLogDetailedInfo(c *C) {
 	}{
 		{
 			sql: "select * from t where t.a <= 15",
-			result: "[stats-feedback] test.t, column: a, range: [-inf,7), actual: 8, expected: 7, buckets: {num: 8 lower_bound: 0 upper_bound: 7 repeats: 1}" +
-				"[stats-feedback] test.t, column: a, range: [8,15), actual: 8, expected: 7, buckets: {num: 8 lower_bound: 8 upper_bound: 15 repeats: 1}",
+			result: "[stats-feedback] test.t, column: a, expected: range: [-inf,7), actual: 8, expected: 7, buckets: {num: 8 lower_bound: 0 upper_bound: 7 repeats: 1}" +
```
Comment (XuHuaiyu) [inline statistics/update_test.go]: The newly added `expected` looks wired.

File: statistics/feedback.go (hunk 8 above)
Comment (qw4990) [reply statistics/feedback.go]: @lamxTyler PTAL

File: statistics/update_test.go (hunk 9 above)
Comment (qw4990) [reply statistics/update_test.go]: updated

File: statistics/update_test.go (hunk 10 above)
Comment (qw4990) [reply statistics/update_test.go]: updated to `rangeStr` now.

File: statistics/feedback.go [hunk 11]
```diff
@@ -1032,7 +1038,7 @@ func dumpFeedbackForIndex(h *Handle, q *QueryFeedback, t *Table) error {
 
 		bytes, err := codec.EncodeKey(sc, nil, ran.LowVal[:rangePosition]...)
 		if err != nil {
-			log.Debug("encode keys failed: err", err)
+			logutil.Logger(context.Background()).Debug("dncoding keys fail", zap.Error(err))
```
Comment (XuHuaiyu) [inline statistics/feedback.go]: s/ dncoding/ encode

File: statistics/feedback.go [hunk 12]
```diff
@@ -1054,7 +1060,7 @@ func dumpFeedbackForIndex(h *Handle, q *QueryFeedback, t *Table) error {
 			continue
 		}
 		if err != nil {
-			log.Debug("get row count by ranges failed: ", err)
+			logutil.Logger(context.Background()).Debug("getting row count by ranges fail", zap.Error(err))
```
Comment (XuHuaiyu) [inline statistics/feedback.go]: It seems we do not need the `ing`s?

Comment (qw4990) [reply statistics/feedback.go]: fixed

File: statistics/feedback.go [hunk 13]
```diff
@@ -944,17 +947,20 @@ func logForIndex(prefix string, t *Table, idx *Index, ranges []*ranger.Range, ac
...
+				zap.String("equality", equalityString), zap.Uint32("expected equality", equalityCount),
+				zap.String("range", rangeString))
 		} else {
 			count, err := getPseudoRowCountByColumnRanges(sc, float64(t.Count), []*ranger.Range{&rang}, 0)
 			if err == nil {
-				log.Debugf("%s index: %s, actual: %d, equality: %s, expected equality: %d, range: %s, pseudo count: %.0f", prefix, idx.Info.Name.O,
-					actual[i], equalityString, equalityCount, rang.String(), count)
+				logutil.Logger(context.Background()).Debug(prefix, zap.String("index", idx.Info.Name.O), zap.Int64("actual", actual[i]),
+					zap.String("equality", equalityString), zap.Uint32("expected equality", equalityCount),
+					zap.String("range", rang.String()), zap.Float64("pseudo count", math.Round(count)))
```
Comment (lysu) [inline statistics/feedback.go]:
```suggestion
					zap.Stringer("range", rang), zap.Float64("pseudo count", math.Round(count)))
```

Comment (qw4990) [reply statistics/feedback.go]: Fixed.

**Comment by qw4990** [comment]
PTAL~

File: statistics/feedback.go [hunk 14]
```diff
@@ -913,18 +916,20 @@ func logForIndexRange(idx *Index, ran *ranger.Range, actual int64, factor float6
 }
 
 func logForIndex(prefix string, t *Table, idx *Index, ranges []*ranger.Range, actual []int64, factor float64) {
+	fmt.Println("=======> log index ", ranges)
```
Comment (winoros) [inline statistics/feedback.go]: remove this

Comment (lamxTyler) [inline statistics/feedback.go]: Remove the debug info?

**Pull request merged**